components/filecoin/rs/src/signature.rs

```diff
@@ -0,0 +1,187 @@
+}
+
+impl Signature {
+    pub fn as_bytes(&self) -> &[u8] {
+        match self {
+            Signature::SignatureSECP256K1(sig_secp256k1) => sig_secp256k1.as_bytes(),
+            Signature::SignatureBLS(sig_bls) => sig_bls.as_bytes(),
+        }
+    }
+}
+
+impl SignatureBLS {
+    pub fn as_bytes(&self) -> &[u8] {
+        &self.0
+    }
+}
+
+impl SignatureSECP256K1 {
+    pub fn as_bytes(&self) -> &[u8] {
+        &self.0
+    }
+}
+
+/// Filecoin Signer Error
+#[derive(Error, Debug)]
+pub enum SignerError {
+    ///  CBOR error
+    #[error("CBOR error: '{0}'")]
+    CBOR(#[from] serde_cbor::Error),
+    /// Secp256k1 error
+    #[error("secp256k1 error")]
+    Secp256k1(#[from] libsecp256k1::Error),
+    // Key decoding error
+    #[error("key decoding error (only hex or base64 is accepted)")]
+    KeyDecoding(),
+    /// Hex Error
+    #[error("Hex decoding error | {0}")]
+    HexDecode(#[from] hex::FromHexError),
+    /// InvalidBigInt error
+    #[error("InvalidBigInt error")]
+    InvalidBigInt(#[from] num_bigint_chainsafe::ParseBigIntError),
+    /// Generic error message
+    #[error("Error: `{0}`")]
+    GenericString(String),
+    /// Not able to parse integer
+    #[error("Cannot parse integer")]
+    ParseIntError(#[from] ParseIntError),
+    /// BLS error
+    #[error("BLS error | {0}")]
+    BLS(#[from] bls_signatures::Error),
+    /// Invalid BIP44Path
+    #[error("Invalid BIP44 path : `{0}`")]
+    InvalidBIP44Path(#[from] zx_bip44::errors::BIP44PathError),
+    /// BLS error
+    #[error("Couldn't convert from slice")]
+    TryFromSlice(#[from] TryFromSliceError),
+    /// Base64 decode Error
+    #[error("Base64 decode error | {0}")]
+    DecodeError(#[from] base64::DecodeError),
+
+    #[error("Marshall error | {0}")]
+    FvmSharedEncodingError(#[from] fvm_ipld_encoding::Error),
+}
+
+const CID_PREFIX: &[u8] = &[0x01, 0x71, 0xa0, 0xe4, 0x02, 0x20];
+const HASH_LENGTH: usize = 32;
+
+/// transform a message into a hashed message ready to be signed and following Filecoin standard
+pub fn get_digest(message: &[u8]) -> Result<[u8; HASH_LENGTH], TryFromSliceError> {
+    let message_hashed =
+        Params::new().hash_length(HASH_LENGTH).to_state().update(message).finalize();
+
+    let cid_hashed = Params::new()
+        .hash_length(HASH_LENGTH)
+        .to_state()
+        .update(CID_PREFIX)
+        .update(message_hashed.as_bytes())
+        .finalize();
+
+    cid_hashed.as_bytes().try_into()
+}
+
+fn transaction_sign_secp56k1_raw(
+    unsigned_message_api: &UnsignedMessage,
+    private_key: &PrivateKey,
+) -> Result<SignatureSECP256K1, SignerError> {
+    let message_cbor = unsigned_message_api.marshal_cbor()?;
+
+    let secret_key = libsecp256k1::SecretKey::parse_slice(&private_key.0)?;
+
+    let cid_hashed = get_digest(message_cbor.as_ref())?;
+
+    let message_digest = Message::parse_slice(&cid_hashed)?;
+
+    let (signature_rs, recovery_id) = sign(&message_digest, &secret_key);
```

Comment:
Can we secret_key after this call here since it will no longer be necessary beyond this function call?

Reply:
done